packages/edit-site/src/index.js

```diff
@@ -57,7 +58,15 @@ export function initializeEditor( id, settings ) {
 	const reboot = reinitializeEditor.bind( null, target, settings );
 
 	dispatch( blocksStore ).__experimentalReapplyBlockTypeFilters();
+
+	addFilter(
+		'blocks.registerBlockType',
+		'core/edit-site/block-register',
+		addInsertersupportForTemplatePartBlock
+	);
 	registerCoreBlocks();
```

Comment:
If this approach sticks, I can see a declarative evolution such as `editor.json` that would look like 
```json
{
    "blockOverrides": {
        "core/template-part": {
            "supports": {
                "inserter": true
            }
        }
    }
}
```

Reply:
I like this idea of different per editor context. I don't want to derail the conversation as this is worth a separate issue, but how would a 3rd-party block achieve the same functionality (working only the site editor) if they can only pack their `block.json` settings but don't have access to the `editor.json` one?